### ISSUE TITLE
GetProductInfoAsync() method not working iOS

### ISSUE THREAD

**Issue by blmiles** [jamesmontemagno/InAppBillingPlugin#349]
I saw a similar issue but that was closed...

I'm trying to use the GetProductInfoAsync() method.
Got the latest great stable version 4.0.2 installed in Xamarin Forms solution in each proj.

Create a simple method in my billing service:

        public async Task<string> GetLocalizedPrice(string prodId)
        {
            var billing = CrossInAppBilling.Current;
            try
            {
                var connected = await billing.ConnectAsync(true);

                string[] prodIds = new string[] { prodId };
                var productfound = await billing.GetProductInfoAsync(ItemType.Subscription, prodIds);
                InAppBillingProduct product = productfound.First();

                return product.LocalizedPrice;
            }
            catch
            {
                return string.Empty;
            }
        }

All other functionality works well. I can check and restore purchases and make a purchase. All subscriptions.
Need to retrieve InAppBillingProduct info as I have different non-std pricing for different countries.

When the GetProductInfoAsync() method is called, the service just hangs, nothing else happens. Never goes further.

And yes, checked prodIDs etc. I'm only passing in one at a time. Same Ids used when checking/restoring or purchasing.

This is targeting the Apple App Store.

Tried in debug mode, in release/Adhoc. Won't retrieve the product info.

Not got to the Google PlayStore just yet but I believe there are issues with this there too.

Any guidance?

Thanks

## Bug Information

Version Number of Plugin: 4.0.2 (stable)
Device Tested On: iPad 5th Gen, iPhoneX
Simulator Tested On:
Version of VS: 2019 - Version 16.8.2
Version of Xamarin: 16.8.000.255
Versions of other things you are using:

### Steps to reproduce the Behavior

See msg above

**Comment by jamesmontemagno** [comment]
hmmm this code is very minimal, I am using it for normal non-consumable products (not subscriptions though)

Can you try to test it out and debug through it by inserting this code in your iOS project:

```
			Task<IEnumerable<SKProduct>> GetProductAsync(string[] productId)
		{
			var productIdentifiers = NSSet.MakeNSObjectSet<NSString>(productId.Select(i => new NSString(i)).ToArray());

			var productRequestDelegate = new ProductRequestDelegate();

			//set up product request for in-app purchase
			var productsRequest = new SKProductsRequest(productIdentifiers)
			{
				Delegate = productRequestDelegate // SKProductsRequestDelegate.ReceivedResponse
			};
			productsRequest.Start();

			return productRequestDelegate.WaitForResponse();
		}

[Preserve(AllMembers = true)]
	class ProductRequestDelegate : NSObject, ISKProductsRequestDelegate, ISKRequestDelegate
	{
		TaskCompletionSource<IEnumerable<SKProduct>> tcsResponse = new TaskCompletionSource<IEnumerable<SKProduct>>();

		public Task<IEnumerable<SKProduct>> WaitForResponse() =>
			tcsResponse.Task;


		[Export("request:didFailWithError:")]
		public void RequestFailed(SKRequest request, NSError error) =>
			tcsResponse.TrySetException(new InAppBillingPurchaseException(PurchaseError.ProductRequestFailed, error.LocalizedDescription));


		public void ReceivedResponse(SKProductsRequest request, SKProductsResponse response)
		{
			var product = response.Products;

			if (product != null)
			{
				tcsResponse.TrySetResult(product);
				return;
			}

			tcsResponse.TrySetException(new InAppBillingPurchaseException(PurchaseError.InvalidProduct, "Invalid Product"));
		}
	}
```

**Comment by jamesmontemagno** [comment]
Also, how are you calling your method: GetLocalizedPrice ? are you using .Result or .Wait?

**Comment by blmiles** [comment]
Hello @jamesmontemagno, thanks for responding.

Actually, the code I posted is fine, all works.
The error was not awaiting the call to my own method.

Once I called my method thus, it all worked: 
var monthlyprice = await billingService.GetLocalizedPrice(prodIdMonthly);
        
I did not have the await AND I was trying to get .Result.
//string monthlyprice = monthlyTask.Result;

Thanks for your efforts, I'll look through your code and pull that in if it'll improve on what I've done.

BTW, really glad you guys finally released a "stable" 4.0.2!
